src/components/Domain/domainTable.tsx

```diff
@@ -23,53 +21,118 @@ import {
 } from "@/components/ui/dialog";
 import { axiosInstance } from "@/lib/axiosinstance";
 import { Button } from "@/components/ui/button";
+import { Switch } from "@/components/ui/switch";
 
 interface DomainData {
   _id: string;
-  domainName: string;
+  label: string;
   description: string;
+  createdAt?: string; // ISO date string
+  createdBy?: string;
+  status?: string; // User or system that created the domain
 }
 
-const truncateText = (text: string, maxLength: number) => {
-  return text.length > maxLength ? text.substring(0, maxLength) + "..." : text;
-};
-
 const DomainTable: React.FC = () => {
   const [domainData, setDomainData] = useState<DomainData[]>([]);
   const [loading, setLoading] = useState(true);
-
-  useEffect(() => {
-    const fetchDomainData = async () => {
-      try {
-        const response = await axiosInstance.get("/domains/all");
-        console.log("API Response:", response.data);
+  const [noData, setNoData] = useState(false); // State to handle no data available
+  const { toast } = useToast();
+  // Function to fetch domain data
+  const fetchDomainData = async () => {
+    setLoading(true);
+    setNoData(false); // Reset noData state before fetching
+    try {
+      const response = await axiosInstance.get("/domain/all");
+      if (response.data.data.length === 0) {
```

Comment:
This traversal 'response.data.data.length' is very dangerous if you don't get response in desired format verify response before traversing properties